Decode/PerfItemType.cs

```diff
@@ -6,90 +6,151 @@ namespace Microsoft.LinuxTracepoints.Decode
     using System.Diagnostics;
 
     /// <summary>
-    /// Provides access to the type of a perf event item. An item is a field of the event
-    /// or an element of an array field of the event. The item may represent one of the
-    /// following, determined by the context that produced this PerfItemType:
+    /// Provides access to the metadata
+    /// of a perf event item. An item is a field of the event or an element of an
+    /// array field of the event.
+    /// <br/>
+    /// The item may represent one of the following, determined by the
+    /// <c>Metadata.IsScalar</c> and <c>Metadata.TypeSize</c>
+    /// properties:
+    ///
     /// <list type="bullet">
+    ///
```

Comment:
Note: Now that we have an `IsScalar` property, it becomes much easier to clearly explain the cases that can occur. As such, the doc comment had to be completely rewritten.